tools/dev/staging-clean.py

```diff
@@ -0,0 +1,58 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+"""
+CodeArtifact does a best practice around immutable versions in
+
+
+for purposes of staging and testing a release, we do want to be able to
+verify the same release after updates/packaging fixes. to do so we
+delete the current package version from the repo.
```

Comment:
If we're treating these staging repos as temporary, would it be a useful option to mint clean namespaced staging repos for builds and then delete them all after a successful release?

I'm not sure if that's more or less work than cleaning out packages individually. If we're iterating through a failure though, wonder if it'd also be useful to refer back to package versions from previous attempts during troubleshooting.

Reply:
yeah, I wondered about that, we currently use a pretty minimal permission set for roles assumed from action, with the code artifact repo provisioned in the infra repo https://github.com/cloud-custodian/infra/tree/main/staging-pypi

Reply:
Ah gotcha, that makes sense 👍